plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
But it is much easier to document "these are all the modules you can register custom stuff with", and have the registration methods document, than have to do that (because we must for non multibinded stuff anyways), *and* have to document "you can also bind these other classes with multi binder".  Also, I think we should minimize how much of guice is exposed in our api, for the same reason we should minimize all dependencies. The less our api relies on it, the easier it could be in the future to switch to something else.